packages/server/src/script/loadRegistrationTesting.ts

```diff
@@ -0,0 +1,142 @@
+import { sampleStateInfo } from '../service/letter'
+import { contactRecords } from '../service/contact'
+import { makeClient, httpConnector } from '@tianhuil/simple-trpc/dist/client'
+import { IVbmRpc, processEnvOrThrow, wait } from '../common'
+import { promises as fs } from 'fs'
+import fetch from 'node-fetch'
+import { TimeoutError } from '@tianhuil/simple-trpc/dist/timedFetch'
+import { RpcRet, IError } from '@tianhuil/simple-trpc/dist/type'
+
+const serverUrl = processEnvOrThrow('REACT_APP_SERVER')
+const timeout = parseInt(processEnvOrThrow('REACT_APP_TIMEOUT'))
+
+export const client = makeClient<IVbmRpc>(
+  httpConnector(serverUrl, { timeout, fetch })
+)
+
+const fakeEmail = 'fake_voter@votebymail.io'
+
+const delay = async <T>(fn: () => Promise<T>, ms: number): Promise<T> => {
+  await wait(ms)
+  return fn()
+}
+
+let timeouts = 0
+let errors = 0
+
+/** Store the amount of timeouts and errors that happened in this test */
+const storeLog = async () => {
+  const file = `${__dirname}/data/loadEmailTesting.log`
+  const content =
+`In the last test ${timeouts} timeouts happened, and ${errors} errors occurred.`
+
+  return fs.writeFile(file, content, { encoding: 'utf-8' })
+}
+
+/**
+ * Store all returned IDs so we can fetch their status later on.
+ *
+ * @param qps The .json file is named after the qps, so we can store the result
+ * of multiple batches simultaneously
+ * @param duration The amount of time in seconds this cycle lasted
+ * @param responses The responses from trpc.register, we parse them to store
+ * registration IDs
+ * @param startTime used to speed up queries, we'll only query events that
+ * happened after this date/time.
+ */
+const storeCycleResults = async (
+  qps: number,
+  duration: number,
+  responses: Array<RpcRet<string> | IError>,
+  startTime: Date,
+) => {
+  const file = `${__dirname}/data/loadEmailTesting_QPS${qps}.json`
+  const ids = responses.map(resp => {
+    if (resp.type === 'data') return resp.data
+    return null
+  }).filter(str => !!str) // removes null elements
+  const content = {
+    total: qps * duration,
+    ids,
+    startTime: startTime.toISOString(),
+  }
+
+  return fs.writeFile(file, JSON.stringify(content), { encoding: 'utf-8' })
+}
+
+/**
+ * Send N amount of registrations per second through a given duration.
+ *
+ * @param qps Amount of registrations to be sent per seconds
+ * @param duration Duration of this cycle in seconds
+ */
+const loadRegistrationTesting = async (qps: number, duration: number) => {
+  const startTime = new Date()
+  const stateInfo = await sampleStateInfo('Florida')
+  stateInfo.email = fakeEmail
+  // Creates an array that has an interval of one second until we reach duration
+  const startTimes = Array(duration).fill(0).map((_, index) => index * 1000)
```

Comment:
This is wrong -- 10 qps for 5 seconds (duration) is 50 queries.

Also, use https://stackoverflow.com/a/37980601/8930600